BaristaSwift/Provider/Provider/View Controller/ViewController.swift

```diff
@@ -140,16 +142,15 @@ class ViewController: UIViewController {
     }
     
     @objc func runRefresh() {
-        self.runSync { () in
-            self.refreshControl.endRefreshing()
-        }
+        _ = self.runSync()
+                .done {
+                    self.updateData()
+                    self.refreshControl.endRefreshing()
+                }
     }
     
-    func runSync(completion:@escaping () -> Void) {
```

Comment:
runSync() used to call updateData(), I moved the updateData() to the done block of callers of runSync().